src/Helper/HeaderHelper.php

```diff
@@ -12,13 +12,30 @@ final class HeaderHelper
      * @return array first element is the value, and key-value are the parameters
      */
     public static function getValueAndParameters(string $headerValue): array
+    {
+        $headerValue = trim($headerValue);
+        if ($headerValue === '') {
+            return [];
+        }
+        $parts = preg_split('/\s*;\s*/', $headerValue, 2, PREG_SPLIT_NO_EMPTY);
+        $output = [$parts[0]];
+        if (count($parts) === 1) {
+            return $output;
+        }
+        return array_merge($output, self::getParameters($parts[1]));
+    }
+
+    /**
+     * Explode header value to parameters (eg. q=2;version=6)
+     */
+    public static function getParameters(string $headerValue): array
     {
         $headerValue = trim($headerValue);
         if ($headerValue === '') {
             return [];
         }
         $parts = preg_split('/\s*;\s*/', $headerValue, -1, PREG_SPLIT_NO_EMPTY);
```

Comment:
I phink this parameters parser is incorrect.\
Please see [this tests](http://test.greenbytes.de/tech/tc2231/#attwithquotedsemicolon) for an example `Content-Disposition` header.
I assume the parsing of the `Content-Disposition: attachment; filename="Here's a semicolon;.html"` will be wrong

Reply:
Maybe we should create a new Issue for detailed testing of the `HeaderHelper` class

Reply:
https://github.com/yiisoft/yii-web/pull/183